Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -3440,54 +3428,46 @@ - (void)sendMessageAttachment:(SignalAttachment *)attachment
+    NSString *temporaryDirectory = NSTemporaryDirectory();
+    NSString *videoDirPath = [temporaryDirectory stringByAppendingPathComponent:@"videos"];
+    if (![[NSFileManager defaultManager] fileExistsAtPath:videoDirPath]) {
+        [[NSFileManager defaultManager] createDirectoryAtPath:videoDirPath
                                   withIntermediateDirectories:YES
                                                    attributes:nil
                                                         error:nil];
     }
-    return [NSURL fileURLWithPath:basePath];
+    return [NSURL fileURLWithPath:videoDirPath];
 }
 
 - (void)sendQualityAdjustedAttachmentForVideo:(NSURL *)movieURL
                                      filename:(NSString *)filename
                            skipApprovalDialog:(BOOL)skipApprovalDialog
 {
+    OWSAssert([NSThread isMainThread]);
+
     AVAsset *video = [AVAsset assetWithURL:movieURL];
     AVAssetExportSession *exportSession =
         [AVAssetExportSession exportSessionWithAsset:video presetName:AVAssetExportPresetMediumQuality];
     exportSession.shouldOptimizeForNetworkUse = YES;
     exportSession.outputFileType = AVFileTypeMPEG4;
-
-    double currentTime = [[NSDate date] timeIntervalSince1970];
-    NSString *strImageName = [NSString stringWithFormat:@"%f", currentTime];
-    NSURL *compressedVideoUrl =
-        [[self videoTempFolder] URLByAppendingPathComponent:[NSString stringWithFormat:@"%@.mp4", strImageName]];
-
+    NSURL *compressedVideoUrl = [[self videoTempFolder]
+        URLByAppendingPathComponent:[[[NSUUID UUID] UUIDString] stringByAppendingPathExtension:@"mp4"]];
     exportSession.outputURL = compressedVideoUrl;
     [exportSession exportAsynchronouslyWithCompletionHandler:^{
-        NSData *videoData = [NSData dataWithContentsOfURL:compressedVideoUrl];
         dispatch_async(dispatch_get_main_queue(), ^{
+            id<DataSource> _Nullable dataSource = [DataSourcePath dataSourceWithURL:compressedVideoUrl];
+            [dataSource setSourceFilename:filename];
             SignalAttachment *attachment =
-                [SignalAttachment attachmentWithData:videoData dataUTI:(NSString *)kUTTypeMPEG4 filename:filename];
+                [SignalAttachment attachmentWithDataSource:dataSource dataUTI:(NSString *)kUTTypeMPEG4];
             if (!attachment || [attachment hasError]) {
-                DDLogWarn(@"%@ %s Invalid attachment: %@.",
+                DDLogError(@"%@ %s Invalid attachment: %@.",
                     self.tag,
                     __PRETTY_FUNCTION__,
                     attachment ? [attachment errorName] : @"Missing data");
                 [self showErrorAlertForAttachment:attachment];
             } else {
                 [self tryToSendAttachmentIfApproved:attachment skipApprovalDialog:skipApprovalDialog];
             }
-
-            NSError *error;
-            [[NSFileManager defaultManager] removeItemAtURL:compressedVideoUrl error:&error];
```

Comment:
We don't need to clean up our temp files ourselves, now that we are creating them in the "temp" folder instead of the "cache" folder.

Reply:
Done.